packages/sdk/src/Incentives/ERC20PeggedVariableCriteriaIncentive.ts

```diff
@@ -0,0 +1,563 @@
+        ...(params as any),
+      },
+    );
+  }
+
+  /**
+   * Fetches the IncentiveCriteria struct from the contract
+   *
+   * @param {?ReadParams} [params]
+   * @returns {Promise<IncentiveCriteria>} Incentive criteria structure
+   * @throws {IncentiveCriteriaNotFoundError}
+   */
+  public async getMaxReward(params?: ReadParams): Promise<bigint> {
+    const maxReward =
+      await readErc20PeggedVariableCriteriaIncentiveGetMaxReward(this._config, {
+        ...params,
+        address: this.assertValidAddress(),
+      });
+
+    return maxReward;
+  }
+
+  /**
+   * The number of claims that have been made
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>}
+   */
+  public async claims(params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveClaims(this._config, {
+      address: this.assertValidAddress(),
+      args: [],
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * The total amount of rewards claimed
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>}
+   */
+  public async totalClaimed(params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveTotalClaimed(
+      this._config,
+      {
+        address: this.assertValidAddress(),
+        args: [],
+        // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+        ...(params as any),
+      },
+    );
+  }
+
+  /**
+   * A mapping of address to claim status
+   *
+   * @public
+   * @async
+   * @param {Address} address
+   * @param {?ReadParams} [params]
+   * @returns {Promise<boolean>}
+   */
+  public async claimed(address: Address, params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveClaimed(this._config, {
+      address: this.assertValidAddress(),
+      args: [address],
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * The address of the ERC20-like token
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<Address>}
+   */
+  public async getAsset(params?: ReadParams) {
+    return await readErc20PeggedVariableCriteriaIncentiveAsset(this._config, {
+      address: this.assertValidAddress(),
+      // biome-ignore lint/suspicious/noExplicitAny: Accept any shape of valid wagmi/viem parameters, wagmi does the same thing internally
+      ...(params as any),
+    });
+  }
+
+  /**
+   * The reward amount issued for each claim
+   *
+   * @public
+   * @async
+   * @param {?ReadParams} [params]
+   * @returns {Promise<bigint>}
+   */
+  public async getReward(params?: ReadParams) {
```

Comment:
These methods 
- getReward
- getLimit

are named just limit and reward in other incentive types. We should keep it consistent.

Reply:
The issue is that it's causing a collision with other params in the file but if it works ship it